bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt

```diff
@@ -311,6 +310,14 @@ class ModelImporter(
             node.setReferenceTarget(it, null as INodeReference?)
         }
     }
+
+    private fun referenceNeedsUpdate(node: INode, referenceRole: String, expectedTargetRef: String): Boolean {
+        val actualTarget = node.getReferenceTarget(referenceRole)
+        val actualTargetRef = actualTarget?.originalId() ?: node.getReferenceTargetRef(referenceRole)?.serialize()
+
+        return actualTargetRef != expectedTargetRef ||
+            actualTarget == null // previously unresolvable reference can now be resolved
```

Comment:
If `actualTarget == null` you can return early and can skip calling `actualTarget?.originalId() ?: node.getReferenceTargetRef(referenceRole)?.serialize()`.

Also, it would be more accurate to say `// Previously unresolvable reference might become resolvable. This is the case when a referenced node is added during the sync.`

Reply:
Your comment made me realize that we still have to check whether it's actually resolvable. With the current implementation we would set an unchanged unresolvable reference to the same unresolvable reference. 
Therefore we would use more operations than we should have.

Reply:
added the additonal check and tested the number of used operations